doc/adr/0003-D3-charts.md

```diff
@@ -0,0 +1,160 @@
+# 3. D3 charts
+
+Date: 2022-02-09
+
+## Status
+
+Proposed.
+
+## Context
+
+ChartJS is a library that facilitates the rendering of data in web applications, but it has limitations.
+
+By using D3 it would be possible to implement all the functionalities currently built with ChartJS, allow the implementation of many more, and reduce code complexity.
+
+This ADR considers the idea and the cost of using D3 instead of ChartJS to render line charts, scatter charts and other types of charts.
+
+### Current limitations
+
+ChartJS does not allow to render HTML elements on top of its canvas. In order to display the vertical blue line that indicates where the user clicked on the chart, a complex implementation involving multiple abstract classes is necessary. Such solution is unnecessarily complex and hard to maintain. D3 allows the integration of features like this with less code and greater simplicity.
+
+Another example are the axis of the charts, which are currently implemented in separate classes from the charts themselves. The y-axis is already implemented with D3. Converting also the charts to D3 would allow a reduced and cleaner code base, which again favors maintainability.
+
+The current implementation of the tooltip is a custom solution that would not be possible to implement with ChartJS. D3 is easier to integrate with custom solutions like this, reducing the time required to develop future improvements.
+
+Small details to improve usability, like highlighting points or lines on hover, are easier to implement with D3. One use case ChartJS failed to deliver recently was highlighting points in the scatter plots when more than one series was plotted. ChartJS allows highlighting only one point.
+
+D3 allows the implementation of small custom details like this one, which enriches the application.
+
+Finally, D3 is a much broader chart library. Implementations beyond the line, scatter, bar and area charts are available and have the potential to improve the application in ways not yet envisioned.
+
+## Decision
+
+Theia Trace Extension already has the D3 library installed. It is currently being used to render the y-axis in some charts. There is no need to install new packages.
+
+As an example, replacing the line and the scatter charts used in `xy-output-component.tsx` requires:
```

Comment:
This file was renamed with a 'proto' prefix since, I believe.

Reply:
In here, the reference regards the real `xy-output-component.tsx` file that is part of the project, not the proto version.